File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        n = len(nums)
        zero_ptr = 0
        for i in range(n):
            if (nums[i] != 0):
                nums[zero_ptr], nums[i] = nums[i], nums[zero_ptr]
                zero_ptr += 1
        

File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int zero_ptr = 0;
        int temp_val = 0;
        for ( int i=0; i<n; i++){
            if (nums[i] != 0){
                temp_val = nums[i];
                nums[i] = nums[zero_ptr];
                nums[zero_ptr] = temp_val;
                zero_ptr++;
            }
        }
            
    }
}